sunpy/net/dataretriever/client.py

```diff
@@ -328,7 +282,12 @@ def fetch(self, qres, path=None, overwrite=False,
         if path is not None:
             path = Path(path)
 
-        urls = [qrblock['url'] for qrblock in qres]
+        urls = []
+        if len(qres):
+            if isinstance(qres, QueryResponseRow):
+                urls = [qres['url']]
```

Comment:
What case (that the tests don't seem to be covering) is this branch of the if statement for?

Reply:
The biggest change made in this PR is that you can now pass rows from the results table through to fetch, i.e. `res = Fido.search(...); Fido.fetch(res['vso'][0])` results in a row of the VSO table being passed through.

Tests are needed, they are still on the TODO.